### DISCUSSION TITLE
What are UserCommands and MessageCommands?

### DISCUSSION THREAD

**Discussion by p10tyr** [discord-net/Discord.Net#2833] [Q&A] [answered]
I don't use Discord much but recently started looking at it and wanted to create my own bot.

Can somebody please update the documentation to include examples of how the text should look like from Discords side.

I have got SlashCommands working with modules and autoregistering, but I have no idea how UserCommands or MessageCommands are supposed to work?

I have spun up the _InteractionFramewWork.Program with the ExampleModule

` [UserCommand("greet")]`
I have tried `!greet @AUserName` in a channel and DM that does not do anything
I tried `/greet @AUserName`  that didn't do anything? 

`[MessageCommand("pin")]`
I tried `!pin a message i tuyped` - Don't seem to get any response or code hits.

Feel like a complete muppet, even looking into the code I have no idea what these are meant to do, how are they differnt from slash commands, etc. 

Any help would be appreciated.. 

Also i can't join discord dnet from the doc site or by finding it on discord. 
https://discord.com/servers/discord-net-848176216011046962

I click accept invite and it just says.. whoops - Unable to accept invite?

Thank you very much

**Comment by Misha-133** [comment, marked as answer]
Message & User commands are different kinds of context menu commands
You can use them from `Apps` submenu in message/user context menus respectively.
They also are application commands, so they're handled just like any other interaction. Here's [a link](https://discordnet.dev/guides/int_framework/intro.html#user-commands) to related docs 

As for the discord server - try this link
https://discord.gg/dnet

**Comment by p10tyr** [comment]
Thank you for explaining that. I can now see what you mean they are under Apps, for the Message. or the User.

![image](https://github.com/discord-net/Discord.Net/assets/1249683/09b7f861-b4ed-4152-a3c2-2a1e2915bf2b)


Still cant join the Discord dnet . same error message

![image](https://github.com/discord-net/Discord.Net/assets/1249683/a3c7d5f3-097e-4b5a-a57f-e4e0cdfb2c87)

![image](https://github.com/discord-net/Discord.Net/assets/1249683/594a0fab-2478-45be-89be-402ef12888cb)

**Reply by Misha-133** [reply]
Hmm. That could happen if you or your alt or someone else's account under the same ip got banned on the server
Try with a VPN ig

**Reply by p10tyr** [reply]
😂😂 Yea. I swtiched to 4G and it got in straight away. I use YouFibre I think they share IP's amongst thousands of users.. super fast but I guess I'll get a static IP later.. Thanks again 👍👍👍👍
